### DISCUSSION TITLE
Regarding adding language support in Dolos API

### DISCUSSION THREAD

**Discussion by AashiqAhmed-Zessta** [dodona-edu/dolos#1187] [Q&A] [answered]
I am using the dolos API running in the docker container. I was trying to compare two swift files but I got the error saying that The module 'tree-sitter-swift' could not be found. Try to install it using npm or yarn, but it may not be supported (yet). So, how do I add the support? I tried adding npm install -g tree-sitter-c-sharp in docker-compose.yaml file, but it didn't work.

**Comment by AashiqAhmed-Zessta** [comment]
Could someone please answer this question? I've attached a screenshot for your reference.

![Screenshot (283)](https://github.com/dodona-edu/dolos/assets/111430494/4415a58d-db92-4ed5-81cd-0e8260166401)

**Comment by rien** [comment, marked as answer]
Hi @AashiqAhmed-Zessta,

When installing additiona `tree-sitter` parsers in the dockerfile, you will need to have the `python3` and `build-base` packages present as well. You can look at the `Dockerfile` for an example:

https://github.com/dodona-edu/dolos/blob/aa03538812f574b826cd2aecb60415c5bd7a5ab9/docker/Dockerfile#L4-L7

**Comment by rien** [comment]
I notice you're using our `docker-compose.yml` setup to run the `Dolos API`. This is primarily made for development purposes and not really meant for production use. I can understand that you want to run Dolos locally, would you want us to make a single Docker container to make this easier for you?

In addition, we can add languages to the official language list of Dolos. This way, you can use our public server as well (https://dolos.ugent.be/server/). Which additional languages do you want us to support?

**Reply by AashiqAhmed-Zessta** [reply]
The languages for which I need support are: go, swift, rust, kotlin and html. We are developing a software used for interview and assessment purposes and we need to integrate dolos in the software for this purpose.

**Reply by rien** [reply]
@AashiqAhmed-Zessta do you still want us to support these additional languages? Since #1291 we can now add new languages much more easily.

In addition, we've also made a production-ready docker-compose setup (see https://dolos.ugent.be/docs/hosting-dolos.html)

**Comment by AashiqAhmed-Zessta** [comment]
Actually, the thing is , we have used the Dolos API for comparing the two files in the same language and store the respective parameters like: similarity, longest fragment etc., in a database.(MongoDB). So, right now, I am using it for local, but we are planning to deploy it to staging and then eventually production later this week.

**Comment by AashiqAhmed-Zessta** [comment]
Ok, for the time being, I'm going to use only the supported languages. Please try adding support for the languages mentioned above.
